### DISCUSSION TITLE
Advice for Networking

### DISCUSSION THREAD

**Discussion by emorling** [fabricjs/fabric.js#6913] [Q&A]
What is your advice for building a networked fabricJS canvas? For example if you want to have 20 people interacting with the canvas at the same time. 

I understand that we can use toObject() and toJSON(), but what is your advice for sending incremental updates?

**Comment by asturur** [comment]
hi @emorling, sending incremental updates isn't a straight forward job, especially if you plan to use a sort of undo/redo.
The easy thing i can imagine is that you add an ID to objects, and then you send only the toObject() of the modified objects.
Then you apply it back on the other side.
You can also look into using `includeDefaultValues` as false, in order to reduce the object output size.

**Reply by emorling** [reply]
That is really good advice, thanks!
